tests/transformer/test_parallel_mlp.py

```diff
@@ -19,28 +21,27 @@ def test_constructor(self, mlp):
         num_weights = sum([p.numel() for p in mlp.parameters()])
         assert num_weights == 1212
 
-    def test_cpu_forward(self, mlp):
+    def test_cpu_forward(self, mlp, transformer_config):
         # [sequence length, micro batch size, hidden size]
-        hidden_states = torch.ones((32, 2, mlp.config.hidden_size))
+        hidden_states = torch.ones((32, 2, transformer_config.hidden_size))
         output, output_bias = mlp(hidden_states)
         assert output.shape[0] == 32
         assert output.shape[1] == 2
-        assert output.shape[2] == mlp.config.hidden_size
-        assert output_bias.shape[0] == mlp.config.hidden_size
+        assert output.shape[2] == transformer_config.hidden_size
+        assert output_bias == None
         assert output.dtype == torch.float32
 
-    @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-    def test_gpu_forward(self, mlp):
-        mlp.cuda()
+    @pytest.mark.skipif(not get_accelerator().is_available(), reason="accelerator not available")
+    def test_accelerator_forward(self, mlp, transformer_config):
+        mlp.to(device_name)
         # [sequence length, batch size, hidden size]
-        hidden_states = torch.ones((32, 2, mlp.config.hidden_size))
-        hidden_states = hidden_states.cuda()
+        hidden_states = torch.ones((32, 2, transformer_config.hidden_size))
+        hidden_states = hidden_states.to(device_name)
         output, output_bias = mlp(hidden_states)
         assert output.shape[0] == 32
         assert output.shape[1] == 2
-        assert output.shape[2] == mlp.config.hidden_size
-        assert output_bias.shape[0] == mlp.config.hidden_size
+        assert output.shape[2] == transformer_config.hidden_size
+        assert output_bias == None
```

Comment:
Why is this change needed?

Reply:
Thanks for the explanation. But should this be conditional based on
https://github.com/microsoft/Megatron-DeepSpeed/blob/7eb36a11b3a9c48ed07b93692ccf22bfb5577f7e/megatron/core/tensor_parallel/layers.py#L484

Reply:
right. but hard to get the value from the test because it's hardcoded in the https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/model/transformer.py#L117. Any ideas?

Reply:
Got it. Thanks for the explanation. I don't see any easy way to resolve this, and I don't think improving this test coverage is worth the trouble.